components/brave_wallet/browser/cardano/cardano_transaction.cc

```diff
@@ -0,0 +1,489 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/cardano/cardano_transaction.h"
+
+#include <algorithm>
+#include <optional>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_wallet/common/cardano_address.h"
+
+namespace brave_wallet {
+
+namespace {
+
+constexpr char kChangeOuputType[] = "change";
+constexpr char kTargetOutputType[] = "target";
+
+bool ReadStringTo(const base::Value::Dict& dict,
+                  std::string_view key,
+                  std::string& to) {
```

Comment:
Please avoid this type of out-param type of thing. These are cases where we should be returning an optional.

Reply:
This form just makes calling code simpler, otherwise it would require a temporary variable, move assignment and else section.

Reply:
I'd prefer this way:
```
  if (!ReadCardanoAddressTo(value, "utxo_address", result.utxo_address)) {
    return std::nullopt;
  }

  if (!ReadDictTo(value, "utxo_outpoint", result.utxo_outpoint)) {
    return std::nullopt;
  }

  if (!ReadUint64StringTo(value, "utxo_value", result.utxo_value)) {
    return std::nullopt;
  }
```
with these functions being in anonymous namespaces and hiding all boilerplate.

Reply:
But my comment was not about these functions, my comment was about `ReadStringTo`.

Reply:
https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs
```Prefer using return values over output parameters: they improve readability, and often provide the same or better performance.```
so this example is also problematic https://github.com/brave/brave-core/pull/28462#discussion_r2034769716 and new code should avoid that. We don't optimize code for the least number of lines or to make it easier to write. We've had previous discussions about consistency with other code and readability for other people. When looking at `ReadCardanoAddressTo` I have no idea that there are out params without looking at the method sig and that hurts readability.

Reply:
Replaced with OptionalUnwrapTo